### PR TITLE
android: clean up AndroidChannelBuilder APIs

### PR DESCRIPTION
**Pull request by voidzcy**
TODO: delete deprecated method after migrating all internal usages.

### CODE REVIEW

File: android/src/main/java/io/grpc/android/AndroidChannelBuilder.java [hunk 1]
```diff
@@ -73,18 +73,43 @@
...
+   */
+  @Deprecated
   public static AndroidChannelBuilder fromBuilder(ManagedChannelBuilder<?> builder) {
     return new AndroidChannelBuilder(builder);
   }
 
+  /**
+   * Creates a new builder, which delegates to the given ManagedChannelBuilder.
+   *
+   * <p>The builder being delegated to can still be modified and configuration changes will
```
Comment (ejona86) [inline android/src/main/java/io/grpc/android/AndroidChannelBuilder.java]: I think we should say, 'The provided {@code builder} becomes "owned" by AndoidChannelBuilder. The caller should not modify the provided builder and AndroidChannelBuilder may modify it.'

File: android/src/main/java/io/grpc/android/AndroidChannelBuilder.java [hunk 2]
```diff
@@ -73,18 +73,43 @@
...
   }
 
+  /**
+   * Creates a new builder, which delegates to the given ManagedChannelBuilder.
+   *
+   * @deprecated Use {@link #usingBuilder(ManagedChannelBuilder)} instead.
+   */
+  @Deprecated
   public static AndroidChannelBuilder fromBuilder(ManagedChannelBuilder<?> builder) {
     return new AndroidChannelBuilder(builder);
```
Comment (dapengzhang0) [inline android/src/main/java/io/grpc/android/AndroidChannelBuilder.java]: Let it explicitly call `usingBuilder()`.

File: android/src/main/java/io/grpc/android/AndroidChannelBuilder.java (hunk 1 above)
Comment (voidzcy) [reply android/src/main/java/io/grpc/android/AndroidChannelBuilder.java]: Is that the expected usage? Later there will be no configuration APIs in `AndroidChannelBuilder` (`transportExecutor()`, `sslSocketFactory()` and `scheduledExecutorService()` will be deleted), so users would not modify `AndroidChannelBuilder` after calling `usingBuilder()` (calling `forTarget()` or `forAddress()` would result in rebuild from scratch, which is not the intended usage). I think users should still be able to modify the provided builder until the channel gets built.

File: android/src/main/java/io/grpc/android/AndroidChannelBuilder.java (hunk 2 above)
Comment (voidzcy) [reply android/src/main/java/io/grpc/android/AndroidChannelBuilder.java]: Done.

File: android/src/main/java/io/grpc/android/AndroidChannelBuilder.java (hunk 1 above)
Comment (ejona86) [reply android/src/main/java/io/grpc/android/AndroidChannelBuilder.java]: I would expect them to change those settings before passing to this builder. Is there a reason they would want to change it later?

Note that it is something we can begin allowing in the future.

Comment (voidzcy) [reply android/src/main/java/io/grpc/android/AndroidChannelBuilder.java]: Ok, that's fair. But `AndroidChannelBuilder` won't modify the provided builder, and users could still use the provided builder, with some modification on configurations, to build another channel. Would the following usage still be legit?

```java
OkHttpChannelBuilder okhttpChBuilder = OkHttpChannelBuilder.forTarget("localhost:8080");
ManagedChannel ch1 = AndroidChannelBuilder.usingBuilder(okhttpChBuilder).build();
ManagedChannel ch2 = AndroidChannelBuilder.usingBuilder(okhttpChBuilder.transportExecutor(...)).build();
```

Comment (voidzcy) [reply android/src/main/java/io/grpc/android/AndroidChannelBuilder.java]: As discussed offline, `AndroidChannelBuilder` should reserve the right to modify the provided builder, even though it currently does not. 

Therefore, the above usage should be discouraged as we do not guarantee `okhttpChBuilder` is not modified after building `ch1`, which may cause unexpected configuration for `ch2`.

Comment (voidzcy) [reply android/src/main/java/io/grpc/android/AndroidChannelBuilder.java]: Done.

**Pull request merged**